src/psyclone/psyGen.py

```diff
@@ -1058,6 +1064,7 @@ def __init__(self, children=None, parent=None):
         else:
             self._children = children
         self._parent = parent
+        self._ast = None  # Reference into fparser2 AST (if any)
```

Comment:
Please add docstring argument information for the method (either in the __init__ or the class docstring).

Reply:
Done. I've also moved the constructor (and `__str__`) up to the start of the class definition as that feels more logical.